docs/performance/inserts/parallel.rst

```diff
@@ -33,35 +34,55 @@ Example
 =======
 
 Suppose we have a stream of data we want to persist into CrateDB.
-You can parallelize this in Java using a
-`CompletableFuture`_ object, like so:
+You can parallelize this in Java using a `CompletableFuture`_ object in order to
+issue insert statements concurrently (non-blocking).
+
+As the java JDBC `Connection`_ is not thread-safe, each insert must use a
+dedicated connection.
+
+A common solution to avoid over-instantiating of connection objects is the usage
+of a pooled JDBC `DataSource`_ like `HikariCP`_, `vibur-dbcp`_, `commons-dbcp`_
+etc.
```

Comment:
```suggestion
A common solution to avoid over-instantiating connection objects is the use of
a pooled JDBC `DataSource`_ like `HikariCP`_, `vibur-dbcp`_, or
`commons-dbcp`_.
```